tools/clang_format.py

```diff
@@ -1,58 +1,56 @@
 #!/usr/bin/env python
 """
-A script that runs clang-format on changes detected via git. It will
-report if running clang-format generated any changes.
+A driver script to run clang-format on changes detected via git.
 
-In CI, the script considers it a failure if running clang-format makes a change.
-In the pre-commit hook, the user is prompted to apply any clang-format changes.
-Running tools/clang_format.py manually with no arguments should replicate the pre-commit hook behavior.
+By default, it will report if running clang-format against lines that
+were changed in the current index generated any changes. If so, it
+exits with non-zero status.
+
+In CI, the script considers it a failure if running clang-format makes
+a change. In the pre-commit hook, the user is prompted to apply any
+clang-format changes. Running tools/clang_format.py manually with no
+arguments should replicate the pre-commit hook behavior.
 
 Only files that are in CLANG_FORMAT_WHITELIST are checked.
 """
-import subprocess
-import os
+
 import argparse
 import difflib
+import os
 import re
+import shlex
+import subprocess
+import sys
 
 
 # Whitelist of directories to check. All files that in that directory
 # (recursively) will be checked.
-CLANG_FORMAT_WHITELIST = ["torch/csrc/jit/", "test/cpp/jit/"]
+CLANG_FORMAT_WHITELIST = [
+    "torch/csrc/distributed/",
+    "torch/lib/c10d/",
```

Comment:
q: is there a way to check which directories are fully compliant (i.e. maybe by running clang-format recursively on them)? If there is, should we do that and add them here?